bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/types/ComplexType.java

```diff
@@ -14,20 +14,24 @@
 
 import java.util.SortedMap;
 
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
 /**
  * A complex type consists out of a sorted list of primitive constituents.
  * Each constituent can be referred to by a unique name.
  *
  * @author Kai Kreuzer - Initial contribution and API
  *
  */
+@NonNullByDefault
 public interface ComplexType extends Type {
 
     /**
      * Returns all constituents with their names as a sorted map
-     * 
+     *
      * @return all constituents with their names
      */
-    public SortedMap<String, PrimitiveType> getConstituents();
+    public SortedMap<String, @Nullable PrimitiveType> getConstituents();

```

Comment:
Perhaps I should remove @Nullable from this Map? See also discussion in https://github.com/eclipse/smarthome/issues/5908.

Reply:
Yes.